File: HAI704I - Architectures Distribuées/TP2/TP2-Java/TP2-Server/src/main/java/tp2/server/model/room/Room.java
package tp2.server.model.room;

import java.awt.Image;
import java.awt.image.BufferedImage;
import java.util.HashSet;
import java.util.Set;

import tp2.server.model.room.bed.Bed;

public class Room
{
	private float price;
	private Image photo;
	private Set<Bed> beds;

	public Room(float price, Bed[] beds)
	{
		this(price, new BufferedImage(1, 1, BufferedImage.TYPE_INT_RGB));
		this.addBeds(beds);
	}

	public Room(float price, Image photo)
	{
		this.price = price;
		this.photo = photo;
		this.beds = new HashSet<Bed>();
	}

	public float getPrice()
	{
		return this.price;
	}
	
	public int getPersonMaxAmount()
	{
		return this.beds.stream().mapToInt(bed -> {
			return bed.getPersonAmount();
		}).sum();
	}

	public void addBeds(Bed[] beds)
	{
		for(Bed bed : beds)
		{
			this.addBed(bed);
		}
	}

	public void addBed(Bed bed)
	{
		this.beds.add(bed);
	}

	public boolean canHost(int persons)
	{
		return this.getPersonMaxAmount() >= persons;
	}
	
	public Image getPhoto()
	{
		return this.photo;
	}
}

File: HAI704I - Architectures Distribuées/TP2/TP2-CS/TP2-Server/TP2-ServiceServer/model/room/Room.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TP2_Server.model.room.bed;

namespace TP2_Server.model.room
{
    public class Room
    {
        public float Price
        {
            get;
        }

        public Image Photo
        {
            get;
        }

        private ISet<IBed> beds;


        public Room(float price, IBed[] beds) : this(price, new Bitmap(1, 1))
        {
            this.AddBeds(beds);
        }

        public Room(float price, Image photo)
        {
            this.Price = price;
            this.Photo = photo;
            this.beds = new HashSet<IBed>();
        }

        public int GetPersonMaxAmount()
        {
            return (from bed in this.beds select bed.GetPersonAmount()).Sum();
        }

        public void AddBeds(IBed[] beds)
        {
            foreach(IBed bed in beds)
            {
                this.AddBed(bed);
            }
        }

        public void AddBed(IBed bed)
        {
            this.beds.Add(bed);
        }

        public bool CanHost(int persons)
        {
            return this.GetPersonMaxAmount() >= persons;
        }
    }
}
